node/node.go

```diff
@@ -415,10 +422,39 @@ func (n *Node) OnStart() error {
 
 	// Always connect to persistent peers
 	if n.config.P2P.PersistentPeers != "" {
-		err = n.sw.DialPeersAsync(n.addrBook, strings.Split(n.config.P2P.PersistentPeers, ","), true)
+		// are any of the persistent peers private?
```

Comment:
it would be great if this logic didn't need to be in `node`. Maybe we can get the `private_peers` field into the switch so it can figure out when to add peers to the addrbook or not. Also the #1244 puts the addrbook in the Switch so at that point we can probably remove it as an arg to DialPeers. 

In any case, this is fine for now, lets open some issues related to p2p cleanup and improving the separation between switch/addrbook/pex